hmcts/pre-api
# Cover suggested error conditions

### JIRA ticket(s)

- S28-3072

### Change description

- Add checks for source container, destination container and source file existing

> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**

- Create a new container in the prefinalsatest (any name is fine)
- Add a sample mp4 video to this new container
- Create a test user using https://pre-api-pr-585.dev.platform.hmcts.net/swagger-ui/index.html
- Call the POST [/media-service/generate-asset](https://pre-api-pr-585.dev.platform.hmcts.net/swagger-ui/index.html#/media-service-controller/generateAsset) endpoint with the app id returned when you created the test user on the previous step and for the code use `SecureKey`. For the body:
```json
{
  "description": "<some test description>",
  "final_asset": "<any unique name>",
  "source_container": "<name of the first container you created>",
  "destination_container": "<name of last container you created>",
  "temp_asset": "<any unique name>"
}
```
- you should expect a 200 result and the destination container should be populated with asset files.
- 404 errors will be returned if the source container doesn't exist or doesn't contain an mp4
